buffer/src/main/java/io/netty/buffer/api/bytebuffer/NioBuffer.java

```diff
@@ -42,19 +42,24 @@
 
 class NioBuffer extends AdaptableBuffer<NioBuffer> implements ReadableComponent, WritableComponent {
     private static final ByteBuffer CLOSED_BUFFER = ByteBuffer.allocate(0);
+    private static final int CLOSED_BUFFER_WMEM_CAP = -1;
 
     private ByteBuffer base;
     private ByteBuffer rmem; // For reading.
     private ByteBuffer wmem; // For writing.
 
     private int roff;
     private int woff;
+    private int rcap;
+    private int wcap;
```

Comment:
not really something we need to do now but we should consider trying to minimise the memory usage by the "NioBuffer container" as otherwise the memory overhead is non-trivial for small allocations. Like we could store rcap and wrap in just one int etc by using masking